### ISSUE TITLE
avdi/naught test suite failing from Module#const_set

### ISSUE THREAD

**Issue by c-lliope** [rubinius/rubinius#2598]
I'm running `rubinius 2.0.0.n255 (1.9.3 a31faf42 2013-09-12 JI) [x86_64-darwin12.4.1]`
on Mac OS 10.8, compiled from commit a31faf4.

I don't know enough metaprogramming to write a minimal failing spec, but here's the failing one from avdi/naught.

Here is [the spec](https://github.com/avdi/naught/blob/master/spec/naught_spec.rb#L87), [the error](https://travis-ci.org/avdi/naught/jobs/11328877#L91), and [the method in the gem](https://github.com/avdi/naught/blob/master/lib/naught/null_class_builder.rb#L40) that's causing the failure.

Basically, it's using `const_set` to create submodules for a class `Foo`, then including those classes into `Foo`. When `Foo.ancestors` is called, it expects the `name` attribute of the superclasses to be the value that was assigned in `const_set`. Instead, the name attributes are nil.

Any thoughts on where I should start digging?
